src/control/client/config.go

```diff
@@ -0,0 +1,195 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/log"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	daosAgentDrpcSockEnv = "DAOS_AGENT_DRPC_DIR"
+	defaultRuntimeDir = "/var/run/daos_agent"
+	defaultLogFile =  "/tmp/daos_agent.log"
+	defaultConfigPath = "etc/daos.yml"
+	defaultSystemName = "daos_server"
+	defaultPort = 10000
+)
+
+// External interface provides methods to support various os operations.
+type External interface {
```

Comment:
There's an existing external interface in the server code that includes getenv. Might be time to move that to the common directory instead of duplicating.

Reply:
Moving the interface to the common appears that it will create a lot of minor edits to the server to make use of the new interface definition in common.  That's doable, but before I do that, I want to make sure we want that change as part of this patch.  Tom, what are you thoughts?

Reply:
I've created a new story for this:  https://jira.hpdd.intel.com/browse/DAOS-2394